includes/collection/class-outbox.php

```diff
@@ -72,6 +79,51 @@ public static function add( $activity_object, $activity_type, $user_id, $content
 			return false;
 		}
 
+		self::invalidate_existing_items( $activity_object->get_id(), $activity_type, $id );
+
 		return $id;
 	}
+
+	/**
+	 * Invalidate existing outbox items with the same activity type and object ID
+	 * by setting their status to 'publish'.
+	 *
+	 * @param string $object_id     The ID of the activity object.
+	 * @param string $activity_type The type of the activity.
+	 * @param int    $current_id    The ID of the current outbox item to exclude.
+	 *
+	 * @return void
+	 */
+	private static function invalidate_existing_items( $object_id, $activity_type, $current_id ) {
+		$meta_query = array(
+			array(
+				'key'   => '_activitypub_object_id',
+				'value' => $object_id,
+			),
+		);
+
+		// For non-Delete activities, only invalidate items of the same type.
+		if ( 'Delete' !== $activity_type ) {
+			$meta_query[] = array(
+				'key'   => '_activitypub_activity_type',
+				'value' => $activity_type,
+			);
+		}
+
+		$existing_items = get_posts(
+			array(
+				'post_type'   => self::POST_TYPE,
+				'post_status' => 'pending',
+				'exclude'     => array( $current_id ),
+				// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+				'meta_query'  => $meta_query,
+				'fields'      => 'ids',
+			)
+		);
+
+		foreach ( $existing_items as $existing_item_id ) {
```

Comment:
```suggestion
		foreach ( $existing_items as $existing_item_id ) {
			$event_args = array(
				Dispatcher::$callback,
				$existing_item_id,
				Dispatcher::$batch_size,
				\get_post_meta( $outbox_item->ID, '_activitypub_outbox_offset', true ) ?: 0, // phpcs:ignore
			);
			$timestamp = wp_next_scheduled( 'activitypub_async_batch', $event_args );
			wp_unschedule_event( $timestamp, 'activitypub_async_batch', $event_args );
```

Reply:
Something like this?

Reply:
nice! I was just sitting on that :)

Reply:
@pfefferle It's untested! I just wrote it down from memory